### DISCUSSION TITLE
Should i set a type for a route response (Typescript)

### DISCUSSION THREAD

**Discussion by AlexPagnotta** [fastify/fastify#3184] [Q&A]
Hi,

I'm using Fastify for a simple project, with typescript and typebox for the schema, and I've some doubts on which is the best practice on define the response type for a route.
I've read the doc, and in some examples a response type is defined, like in the example below, sometimes it's not, so it's not clear to me if there is a real advantage in defining it, as it seems that there is no check when we do reply.send() with a "wrong" type.

` app.post<{ Body: UserType; Response: UserType }>(...`

Thanks!

P.S It's possible to define only the response type, omitting body, params etc?
Something like that:

` app.post<{ Response: UserType }>(...`

**Comment by carloscarnero** [comment]
For simple JSON responses, what follows is what I'm currently doing. It does seem to work; however, I don't know if there's a better method. First, the response body type:

```TypeScript
type ResponseDocument = {
  message: string;
};
```

and the handler:

```TypeScript
fastify.get("/test", async (request, reply): Promise<ResponseDocument> => {
  return {
    message: "Hello, world!"
  }
});
```
Notice the function's return type.

**Comment by shwao** [comment]
I made a pull request to update the docs. I believe there is a mistake. It should be `Reply` not `Response`. Reply works.

https://github.com/fastify/fastify/pull/3222
